packages/aws-cdk-lib/aws-ssm/lib/parameter.ts

```diff
@@ -574,12 +574,17 @@ export class StringParameter extends ParameterBase implements IStringParameter {
    *
    * Requires that the stack this scope is defined in will have explicit
    * account/region information. Otherwise, it will fail during synthesis.
+   *
+   * If defaultValue is provided, it will be used as the dummyValue
+   * and the ContextProvider will be told NOT to raise an error on synthesis
+   * if the SSM Parameter is not found in the account at synth time.
    */
-  public static valueFromLookup(scope: Construct, parameterName: string): string {
+  public static valueFromLookup(scope: Construct, parameterName: string, defaultValue?: string): string {
     const value = ContextProvider.getValue(scope, {
       provider: cxschema.ContextProvider.SSM_PARAMETER_PROVIDER,
       props: { parameterName },
-      dummyValue: `dummy-value-for-${parameterName}`,
+      dummyValue: defaultValue || `dummy-value-for-${parameterName}`,
+      ignoreErrorOnMissingContext: !!defaultValue,
```

Comment:
```suggestion
      ignoreErrorOnMissingContext: defaultValue !== undefined,
```

If the default value is the empty string, this will turn it into `false`, but you probably expect it to be `true`.

Reply:
Good call.  And you're right, I do.